### PR TITLE
`Development`: Assign latest docker tag to release and develop tag for commits to dev

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Currently the `:latest` docker tag is updated on every commit to the develop branch. Since docker defaults to the latest tag and many docker projects use latest as their release tag, we should do the same. If someone wants to have the latest dev build, he can now use the `develop` tag.

### Description

This PR changes this behavior, so every commit to the develop branch will update the `develop` tag. The latest tag will only be updated on a new release. 

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
Code logic looks good

**Review by bassner — APPROVED** [review]
Looks good to me.

**Review by Hialus — APPROVED** [review]
Changes will do what you described. 👍🏻

File: .github/workflows/build-deploy.yml [hunk 1]
```diff
@@ -98,12 +98,15 @@ jobs:
           if (context.eventName === "pull_request") {
             return "pr-" + context.issue.number;
           }
+          if (context.eventName === "release") {
```
Comment (4ludwig4) [inline .github/workflows/build-deploy.yml:101]: Do we release Hotfixes for old versions? Or are we always increasing the version number? In the hotfix situation of a previous major version we would have problems with setting latest just by checking for a release type.

**Review by 4ludwig4 — APPROVED** [review]
Code looks good to me. The one comment I had is probably an edge case. Also that edge case probably already exists also in the current version.

File: .github/workflows/build-deploy.yml (hunk 1 above)
Comment (TheZoker) [reply .github/workflows/build-deploy.yml:101]: I think that just a increased number will be used then. 
@krusche What's your thoughts on this?

Comment (sleiss) [reply .github/workflows/build-deploy.yml:101]: According to the Security Policy, we do not provide fixes for old versions (and I also have never seen such fix).
https://github.com/ls1intum/Artemis/blob/develop/SECURITY.md

**Review by Mtze — APPROVED** [review]

**Review by sleiss — APPROVED** [review]

**Review by manuelmanso — APPROVED** [review]

**Pull request merged**